dsalgo/linked_list.py

```diff
@@ -331,53 +331,76 @@ def skip_i_delete_j(self, i, j):
 		return - return the updated head of the linked list
 		"""
 		'''
-		The Idea: 
+		The Idea:
 			Traverse the Linkedist. Make use of two references - `current` and `previous`.
-			- Skip `i-1` nodes. Keep incrementing the `current`. Mark the `i-1`^th node as `previous`. 
+			- Skip `i-1` nodes. Keep incrementing the `current`. Mark the `i-1`^th node as `previous`.
 			- Delete next `j` nodes. Keep incrementing the `current`.
 			- Connect the `previous.next` to the `current`
 		'''
 		# Edge case - Skip 0 nodes (means Delete all nodes)
 		head = self.head
 		if i == 0:
 			return None
-		
+
 		# Edge case - Delete 0 nodes
 		if j == 0:
 			return head
-		
+
 		# Invalid input
 		if head is None or j < 0 or i < 0:
 			return head
 
 		# Helper references
 		current = head
 		previous = None
-		
+
 		# Traverse - Loop untill there are Nodes available in the LinkedList
 		while current:
-			
+
 			'''skip (i - 1) nodes'''
 			for _ in range(i - 1):
 				if current is None:
 					return head
 				current = current.next
 			previous = current
 			current = current.next
-			
+
 			'''delete next j nodes'''
 			for _ in range(j):
 				if current is None:
 					break
 				next_node = current.next
 				current = next_node
-			
-			'''Connect the `previous.next` to the `current`''' 
+
+			'''Connect the `previous.next` to the `current`'''
 			previous.next = current
-		
+
 		# Loop ends
-		
+
 		return head
+
+
+	@staticmethod
+	def sorted_merge(list1, list2):
+
+		"""
+		:param: head - head of first linked list
+		:param: list2 - head of second linked list
+		return - return a new sorted merged list
+		"""
+
+		sorted_array = []
+		array = list1.to_list()
+		linked_list = LinkedList()
+		for value in array:
+			linked_list.sort_append(value)
+		array = list2.to_list()
+		for value in array:
+			linked_list.sort_append(value)
+
+		return linked_list
+
 		
 	def __repr__(self):
 		return "LinkedList({})".format((self.to_list()))
+
```

Comment:
define the sorted_merge function explicitly i.e. not in the class.
and also don't remove this function and use the self instead of list1.
so this is the final instructions:-
- in the current sorted_merge function use self instead of list1.
- define a new function sorted merge outside the class which takes two args list1 and list2 and returns merged linked list.